template.go

```diff
@@ -52,8 +52,8 @@ func (t template) Write(ctx context.Context, f *Function) error {
 	if f.Deploy.HealthEndpoints.Readiness == "" {
 		f.Deploy.HealthEndpoints.Readiness = t.config.HealthEndpoints.Readiness
 	}
-	if f.Invocation.Format == "" {
-		f.Invocation.Format = t.config.Invocation.Format
+	if f.Invoke == "" && t.config.Invoke != DefaultInvokeFormat {
```

Comment:
```suggestion
	if f.Invoke == "" && t.config.Invoke != "http" {
```
I would change this to the actual value, since that is "locked in time" as being the value pre-migration-35, whereas it is possible (though very unlikely) that we could change DefaultInvokeFormat in the future